addons/website_sale_stock/views/website_sale_stock.xml

```diff
@@ -31,4 +39,35 @@
       </t>
     </xpath>
   </template>
-</odoo>
\ No newline at end of file
+
+  <template id="products_out_of_stock" inherit_id="website_sale.products_add_to_cart" name="Out of Stock Products">
+    <xpath expr="//a[@class='btn btn-default btn-xs a-submit']" position="replace">
+        <a t-attf-class="btn btn-default btn-xs a-submit #{ 'disabled' if website.stock_warning_active and product.qty_available&lt;= 0 and product.type=='product' else '' }">
+            <span class="fa fa-shopping-cart"/>
+        </a>
+    </xpath>
+  </template>
+
+  <template id="product_out_of_stock" inherit_id="website_sale.product" name="Out of Stock Product">
+    <xpath expr="//a[@id='add_to_cart']" position="replace">
```

Comment:
don't replace the complete button. Just add your input with `data-stock` after the input `name="product_id"`.